lib/Target/ExportSMTLIB/ExportSMTLIB.cpp

```diff
@@ -334,9 +337,21 @@ struct ExpressionVisitor
     newInfo.stream.indent(newInfo.indentLevel);
     if (failed(printExpression(worklist, newInfo)))
       return failure();
+
     info.stream << info.valueMap.lookup(yieldedValue);
-    for (unsigned k = 0; k < newInfo.openParens + 1; ++k)
-      info.stream << ")";
+
+    if(weight != 0 && newInfo.openParens > 0){
+      for(int j=0;j<newInfo.openParens;j++)
+        info.stream <<")";
+      info.stream <<" :weight "<< weight << "))";
+    } else if (weight != 0){
+      info.stream <<" :weight "<< weight << "))";
+      for(int j=0;j<newInfo.openParens;j++)
+        info.stream <<")";
+    } else{
+      for (unsigned k = 0; k < newInfo.openParens+1; ++k)
+        info.stream << ")";
```

Comment:
Nit: This and the loop in the first branch could be factored out and be before the if statement chain with `newInfo.openParens` as the upper bound. Then we only have to print one closing parenthesis here. Also, the first two branches would be the same then and could be merged.